language_service/src/format.rs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use crate::{compilation::Compilation, protocol::TextEdit};
+
+use qsc::formatter::calculate_format_edits;
+use qsc::line_column::{Encoding, Position, Range};
+
+pub(crate) fn get_format_changes(
+    compilation: &Compilation,
+    source_name: &str,
+    _position: Position,
+    encoding: Encoding,
+) -> Vec<TextEdit> {
+    let contents = compilation
+        .user_unit()
+        .sources
+        .find_by_name(source_name)
+        .expect("can't find source by name")
+        .contents
+        .clone();
```

Comment:
Is the `.clone()` necessary?

Reply:
Normally yes, but since `contents` is only used by reference later on, we can omit the clone here in favor of making a reference here.